web/frontend/src/components/utils/useChangeAction.tsx

```diff
@@ -169,69 +172,99 @@ const useChangeAction = (
   };
 
   const getAction = () => {
```

Comment:
This is good enough for now as it is legacy code. But instead of returning a single action based on the role and status, each action should be rendered based on those two parameters. Like so we would have each component defined separately, and we could have not only one action. For example:

```ts
const OpenBTN = ({status}) => {
  const { authCtx } = useContext(AuthContext);

  const isAuthorized = () : boolean {
    return authCtx.role === 'admin' || authCtx.role === 'operator'
  }

  return (
    {isAuthorized && status === STATUS.INITIAL ? (
      <button>...</button>
    ) : null}
  )
}

const CloseBTN = ({status}) => {
  const { authCtx } = useContext(AuthContext);

  const isAuthorized = () : boolean {
    return authCtx.role === 'admin' || authCtx.role === 'operator'
  }

  return (
    {isAuthorized && status === STATUS.OPEN ? (
      <button>...</button>
    ) : null}
  )
}
```

And then, in the main page you have:

```ts
  <div>
    ...
    <OpenBTN status={status} />
    <CloseBTN status={status} />
    ...
  </div>
```